### PR TITLE
Kernel: Store the full git hash inside the kernel

### LINKED ISSUE #6: API for getting git revision should be more robust.
If I understood @Cleverking2003 correctly, currently OS revision is stored in 32-bit integer which only allows to store first 4 bytes of 20-byte git hash.

This should be reworked.

### PR DESCRIPTION
**Pull request by Cleverking2003**
image

Closes #6

### CODE REVIEW

**Review by mkostoevr — CHANGES_REQUESTED** [review]
Please update the kernel_revision test ([kernel/test/kernel_revision/test.py +9](https://github.com/kolibrios-nextgen/kolibrios-ng/blob/main/kernel/test/kernel_revision/test.py#L9)).

**Review by mkostoevr — APPROVED** [review]

File: kernel/Tupfile.lua [hunk 1]
```diff
@@ -1,9 +1,9 @@
 if tup.getconfig("NO_FASM") ~= "" then return end
 tup.rule("echo lang fix " .. ((tup.getconfig("LANG") == "") and "en" or tup.getconfig("LANG")) .. " > %o", {"lang.inc"})
-tup.rule("echo __REV = 0x`git rev-parse --short HEAD` > %o", {"rev.inc"})
-tup.rule({"bootbios.asm", extra_inputs = {"lang.inc", "rev.inc"}}, "fasm %f %o ", "bootbios.bin")
-tup.rule({"bootbios.asm", extra_inputs = {"lang.inc", "rev.inc"}}, "fasm %f %o -dextended_primary_loader=1", "bootbios.bin.ext_loader")
-tup.rule({"bootbios.asm", extra_inputs = {"lang.inc", "rev.inc"}}, "fasm %f %o -dpretest_build=1", "bootbios.bin.pretest")
+tup.rule("echo -n `git rev-parse HEAD` > %o", {"rev.inc"})
```
Comment (mkostoevr) [inline kernel/Tupfile.lua:3]: Can this be simplified as `git rev-parse HEAD > %o`?

File: kernel/Tupfile.lua [hunk 2]
```diff
@@ -1,9 +1,9 @@
 if tup.getconfig("NO_FASM") ~= "" then return end
 tup.rule("echo lang fix " .. ((tup.getconfig("LANG") == "") and "en" or tup.getconfig("LANG")) .. " > %o", {"lang.inc"})
-tup.rule("echo __REV = 0x`git rev-parse --short HEAD` > %o", {"rev.inc"})
-tup.rule({"bootbios.asm", extra_inputs = {"lang.inc", "rev.inc"}}, "fasm %f %o ", "bootbios.bin")
```
Comment (mkostoevr) [inline kernel/Tupfile.lua:4]: I've lost, why do we remove the "rev.inc" dependency? Won't it cause the system to not make rev.inc in clean build?

File: kernel/kernel.asm [hunk 3]
```diff
@@ -2474,8 +2470,7 @@ endg
 iglobal
 version_inf:
         db 0,7,7,0  ; version 0.7.7.0
-        db 0
-.rev    dd __REV__
+.rev    file "rev.inc"
```
Comment (mkostoevr) [inline kernel/kernel.asm:2473]: Please only include 40 bytes of the file here (`file "rev.inc":0,40`) and create the "rev.inc" using a regular shell redirect.

File: kernel/Tupfile.lua (hunk 1 above)
Comment (mkostoevr) [inline kernel/Tupfile.lua:3]: Please use a regular shell redirect here (`git rev-parse HEAD > %o`), it will make the rule valid for windows.

**Review by mkostoevr — CHANGES_REQUESTED** [review]

File: kernel/Tupfile.lua (hunk 2 above)
Comment (Cleverking2003) [reply kernel/Tupfile.lua:4]: That's because the bootloader doesn't depend on `rev.inc` anymore, it is only used in the kernel.

**Review by mkostoevr — APPROVED** [review]
Thanks!

Looks great.

**Pull request merged**